### DISCUSSION TITLE
When will JDA version 5 be released?

### DISCUSSION THREAD

**Discussion by gaborszita** [discord-jda/JDA#2226] [Questions and Help] [answered]
When will the stable JDA version 5, or at least the beta of it, be released? I really need some features in this new release, but since there are [breaking changes](https://github.com/DV8FromTheWorld/JDA/discussions/2156) in the alpha version, I can't use the alpha. I don't need the exact date for it, but is there an ETA? Will it be 2 months? 1 year? I don't necessarily need a stable or beta version - I just need a release after which their won't be any more breaking changes in v5 of the API, because obviously I don't want to refactor everything when I just upgrade to a newer release.

**Comment by MinnDevelopment** [comment, marked as answer]
It will be released when it's done. There is no ETA.

Breaking changes will *always* happen even after the alpha is over. That is simply the nature of discord libraries, which only wrap an API which itself receives changes over time.

The general recommendation is, and always has been, to update to v5 if you need the features. It being alpha does not make it unusable. Many people are using it in production for a while already. Keep in mind that breaking changes only affect you when you update the dependency in the first place.

**Reply by gaborszita** [reply]
> It will be released when it's done. There is no ETA.
> 
> Breaking changes will _always_ happen even after the alpha is over. That is simply the nature of discord libraries, which only wrap an API which itself receives changes over time.
> 
> The general recommendation is, and always has been, to update to v5 if you need the features. It being alpha does not make it unusable. Many people are using it in production for a while already. Keep in mind that breaking changes only affect you when you update the dependency in the first place.

Thanks for your quick reply. Do you have a list of features that are yet to be implemented or are already implemented? Or something that shows the progress of v5?

**Reply by MinnDevelopment** [reply]
The already implemented features are all listed in the release changelogs: https://github.com/DV8FromTheWorld/JDA/releases

Any future features are always in the pull requests: https://github.com/DV8FromTheWorld/JDA/pulls

To help with migration there are also a few helpful migration tips in the #jda5-changes channel on our discord server at https://discord.gg/jda, and you can also ask questions in the help forum there.
